### PR TITLE
Status event log 42

### PR DESCRIPTION
**Pull request by JacksonM8**
Support for per-process logging, bug fix for multiple logging, backwards compatibility for interchangable  pqxx::row & pqxx::tuple usage

### CODE REVIEW

File: src/aggregationserver/databaseclient.h [hunk 1]
```diff
@@ -9,6 +9,13 @@
 
 #include <iostream>
 
+// Hack for backwards compatibility with pqxx 4.0 and earlier where row was named tuple
+#ifndef PQXX_H_ROW
+namespace pqxx{
+    typedef tuple row;
```
Comment (mitchell-conrad) [inline src/aggregationserver/databaseclient.h]:
```suggestion
    using row = tuple;
```

File: src/aggregationserver/databaseclient.cpp [hunk 2]
```diff
@@ -68,7 +68,7 @@ int DatabaseClient::InsertValues(const std::string& table_name,
         query_stream << connection_.quote(values.at(i)) << ',';
     }
     query_stream << connection_.quote(values.at(values.size()-1)) << ")" << std::endl;
-    query_stream << "RETURNING " << id_column << ";" << std::endl;
+    query_stream << "RETURNING " << /*table_name << "ID"*/ strip_schema(table_name) << "." << id_column << ";" << std::endl;
```
Comment (mitchell-conrad) [inline src/aggregationserver/databaseclient.cpp]:
```suggestion
    query_stream << "RETURNING " << strip_schema(table_name) << "." << id_column << ";" << std::endl;
```

File: src/aggregationserver/databaseclient.cpp [hunk 3]
```diff
@@ -492,12 +501,16 @@ const std::string DatabaseClient::BuildColTuple(const std::vector<std::string>&
 
 pqxx::work& DatabaseClient::AquireBatchedTransaction() {
     std::lock_guard<std::mutex> trans_guard(batched_transaction_mutex_);
+    static unsigned int total_batched_bois = 0;
 
-    if (batched_transaction == nullptr) {
-        batched_transaction = std::unique_ptr<pqxx::work>(new pqxx::work(batched_connection_, "BatchedTransaction"));
+    if (batched_transaction_ == nullptr) {
+        batched_transaction_ = std::unique_ptr<pqxx::work>(
+            new pqxx::work(batched_connection_, "BatchedTransaction"+std::to_string(total_batched_bois))
```
Comment (mitchell-conrad) [inline src/aggregationserver/databaseclient.cpp]:
```suggestion
            new pqxx::work(batched_connection_, "BatchedTransaction"+std::to_string(total_batched_transactions_))
```

File: src/aggregationserver/databaseclient.cpp [hunk 4]
```diff
@@ -492,12 +501,16 @@ const std::string DatabaseClient::BuildColTuple(const std::vector<std::string>&
 
 pqxx::work& DatabaseClient::AquireBatchedTransaction() {
     std::lock_guard<std::mutex> trans_guard(batched_transaction_mutex_);
+    static unsigned int total_batched_bois = 0;
```
Comment (mitchell-conrad) [inline src/aggregationserver/databaseclient.cpp]:
```suggestion
    static unsigned int total_batched_transactions_ = 0;
```

File: src/aggregationserver/databaseclient.cpp [hunk 5]
```diff
@@ -492,12 +501,16 @@ const std::string DatabaseClient::BuildColTuple(const std::vector<std::string>&
...
 
-    if (batched_transaction == nullptr) {
-        batched_transaction = std::unique_ptr<pqxx::work>(new pqxx::work(batched_connection_, "BatchedTransaction"));
+    if (batched_transaction_ == nullptr) {
+        batched_transaction_ = std::unique_ptr<pqxx::work>(
+            new pqxx::work(batched_connection_, "BatchedTransaction"+std::to_string(total_batched_bois))
+        );
         batched_write_count_ = 0;
+        total_batched_bois++;
```
Comment (mitchell-conrad) [inline src/aggregationserver/databaseclient.cpp]:
```suggestion
        total_batched_transactions_++;
```

File: src/aggregationserver/databaseclient.cpp [hunk 6]
```diff
@@ -510,6 +523,26 @@ void DatabaseClient::ReleaseBatchedTransaction() {
...
-    batched_transaction->commit();
-    batched_transaction.reset();
+    if (batched_transaction_ == nullptr) {
+        return;
+    }
+    try {
+        batched_transaction_->commit();
+    } catch (const std::exception& e) {
+        std::cerr << "An exception occurred while attempting to commit a batched transaction" << std::endl;
+        std::cerr << "Pre-release batched transaction: " << batched_transaction_.get() << std::endl;
```
Comment (mitchell-conrad) [inline src/aggregationserver/databaseclient.cpp]:
```suggestion
```

File: src/aggregationserver/databaseclient.cpp [hunk 7]
```diff
@@ -510,6 +523,26 @@ void DatabaseClient::ReleaseBatchedTransaction() {
...
+        batched_transaction_.reset();
+        std::cerr << "Released batched transaction: " << batched_transaction_.get() << std::endl;
```
Comment (mitchell-conrad) [inline src/aggregationserver/databaseclient.cpp]:
```suggestion
```

File: src/aggregationserver/experimenttracker.cpp [hunk 8]
```diff
@@ -21,11 +21,14 @@ ExperimentTracker::ExperimentTracker(std::shared_ptr<DatabaseClient> db_client)
     
     }
 
-int ExperimentTracker::RegisterExperimentRun(const std::string& experiment_name, double timestamp) {
+int ExperimentTracker::RegisterExperimentRun(
+    const std::string& experiment_name,
+    const google::protobuf::Timestamp& timestamp
+) {
     // TODO: quick fix for demo, should drive time value off of message once it is being populated
```
Comment (mitchell-conrad) [inline src/aggregationserver/experimenttracker.cpp:28]: Create ticket to investigate this?

File: src/aggregationserver/systemeventprotohandler.cpp [hunk 9]
```diff
@@ -29,13 +29,114 @@ void SystemEventProtoHandler::ProcessStatusEvent(const SystemEvent::StatusEvent&
...
+    const auto& phys_mem_util = std::to_string(p_status.phys_mem_utilization());
+    const auto& phys_mem_used_kb = std::to_string(p_status.phys_mem_used_kb());
+
+    const auto& threads = std::to_string(p_status.thread_count());
+    const auto& disk_read_kb = std::to_string(p_status.disk_read_kilobytes());
+    const auto& disk_write_kb = std::to_string(p_status.disk_written_kilobytes());
+    const auto& disk_total_kb = std::to_string(p_status.disk_total_kilobytes());
+
+    const auto& cpu_time = TimeUtil::ToString(p_status.cpu_time());
+    const std::string& state = SystemEvent::ProcessStatus::State_Name(p_status.state());
```
Comment (mitchell-conrad) [inline src/aggregationserver/systemeventprotohandler.cpp]:
```suggestion
    const auto& state = SystemEvent::ProcessStatus::State_Name(p_status.state());
```

**Review by mitchell-conrad — CHANGES_REQUESTED** [review]
Looks good, few suggestions and changes.
Table insert vector size checks might be good.

File: src/aggregationserver/experimenttracker.cpp (hunk 8 above)
Comment (JacksonM8) [reply src/aggregationserver/experimenttracker.cpp:28]: Added; LOG-106

**Pull request merged**